### PR TITLE
Text / Heading: Add semibold weight & introduce codemod unit tests

### PR DESCRIPTION
**Pull request by christianvuerings**
Adds a semi bold option to `<Text />` and `<Heading />`

image

- [*] Documentation
- [*] Tests
- [*] Accessibility checkup

We also introduce codemod unit tests to more easily write codemods, check edge cases & avoid regressions.

More info at
- https://github.com/facebook/jscodeshift#unit-testing
- https://github.com/benjamn/recast
- https://github.com/facebook/jscodeshift
- https://github.com/cpojer/js-codemod
- https://github.com/reactjs/react-codemod

### CODE REVIEW

File: .flowconfig [hunk 1]
```diff
@@ -5,6 +5,7 @@
 # Tracking https://github.com/facebook/flow/issues/4015
 <PROJECT_ROOT>/docs
 <PROJECT_ROOT>/test
+<PROJECT_ROOT>/packages/gestalt-codemods/.*/__testfixtures__
```
Comment (christianvuerings) [inline .flowconfig]: Flow fails on `<Text bold />` referenced in test fixtures, we could add something like `$FlowExpectedError`, however that breaks the `input` / `output` behavior `defineTest` expects

File: packages/gestalt-codemods/0.108.0-0.109.0/__testfixtures__/convert-font-weight-dynamic.input.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+// @flow
```
Comment (christianvuerings) [inline packages/gestalt-codemods/0.108.0-0.109.0/__testfixtures__/convert-font-weight-dynamic.input.js]: More info on naming requirements are at https://github.com/facebook/jscodeshift#unit-testing

> jscodeshift comes with a simple utility to allow easy unit testing with [Jest](https://facebook.github.io/jest/), without having to write a lot of boilerplate code. This utility makes some assumptions in order to reduce the amount of configuration required:
> - The test is located in a subdirectory under the directory the transform itself is located in (eg. `__tests__`)
> - Test fixtures are located in a `__testfixtures__` directory

File: packages/gestalt-codemods/0.108.0-0.109.0/__testfixtures__/.eslintrc [hunk 3]
```diff
@@ -0,0 +1,8 @@
+{
+  "rules": {
+    "prettier/prettier": 0,
```
Comment (christianvuerings) [inline packages/gestalt-codemods/0.108.0-0.109.0/__testfixtures__/.eslintrc]: Had to disable prettier in the test fixture because of
```
<Text size="xl" weight={isCurrentPage ? "bold" : "normal"}>
```
should instead be
```
<Text size="xl" weight={isCurrentPage ? 'bold' : 'normal'}>
```

However, when calling jscodeshift with `.toSource({quote: 'single'})`, it converts all quotes instead of only the modified ones.

File: packages/gestalt-codemods/0.108.0-0.109.0/__tests__/convert-font-weight.test.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
+import { defineTest } from 'jscodeshift/dist/testUtils.js';
+
+jest.mock('../convert-font-weight', () => {
+  return Object.assign(require.requireActual('../convert-font-weight'), {
```
Comment (christianvuerings) [inline packages/gestalt-codemods/0.108.0-0.109.0/__tests__/convert-font-weight.test.js]: Found this fix in `react/react-codemods`: https://git.io/JePID

File: README.md [hunk 5]
```diff
@@ -61,20 +61,15 @@ Running Masonry's integration tests. This will leave lots of Firefox processes h
 
 When a release will cause breaking changes — in usage or in typing — we provide a codemod to ease the upgrade process. Codemods are organized by release in `/packages/gestalt-codemods`. We recommend using [jscodeshift](https://github.com/facebook/jscodeshift) to upgrade.
 
-### Prerequisite:
-Install `jscodeshift` globally if you haven't already.
-```bash
-yarn global add jscodeshift
```
Comment (christianvuerings) [inline README.md]: `jscodeshift` is now installed as a dev dependency and we wrap it with `yarn codemod` to ensure the correct `jscodeshift` version.

File: packages/gestalt-codemods/0.108.0-0.109.0/__tests__/convert-font-weight.test.js [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+    parser: 'flow',
+  });
+});
+
+describe('convert-font-weight', () => {
+  [
+    'convert-font-weight-dynamic',
+    'convert-font-weight-spread',
+    'convert-font-weight-static',
```
Comment (btraut) [inline packages/gestalt-codemods/0.108.0-0.109.0/__tests__/convert-font-weight.test.js]: Do we also need to worry about any cases with `<Text bold={false} />`?

File: packages/gestalt/src/Heading.js [hunk 7]
```diff
@@ -74,15 +76,16 @@ export default function Heading(props: Props) {
     lgSize && styles[`lgFontSize${SIZE_SCALE[lgSize]}`],
     colors[color],
     overflow === 'breakWord' && typography.breakWord,
-    truncate && typography.truncate
+    truncate && typography.truncate,
+    weight === 'semibold' && typography.fontWeightSemibold
```
Comment (btraut) [inline packages/gestalt/src/Heading.js]: Do you tend to like to make secondary properties like `fontWeightSemibold` as just overrides to the base class? I will often create two separate classes each that extend the base, then choose which one of those I should use. I'm certain your approach works, but I'm curious what you think, and if you think I'm being too pedantic.

File: packages/gestalt-codemods/0.108.0-0.109.0/__tests__/convert-font-weight.test.js (hunk 6 above)
Comment (christianvuerings) [reply packages/gestalt-codemods/0.108.0-0.109.0/__tests__/convert-font-weight.test.js]: Thanks - also added a test case for that (didn't find any in pinboard)

File: packages/gestalt/src/Heading.js (hunk 7 above)
Comment (christianvuerings) [reply packages/gestalt/src/Heading.js]: @btraut this approach is a bit more atomic, just one property per CSS class and also allows for better re-use across components.
